ISPN-12868 Configure ACL cache

tristantarrant opened the pull request:
https://issues.redhat.com/browse/ISPN-12868

gustavonalle commented on core/src/main/java/org/infinispan/security/GlobalSecurityManager.java:
```diff
@@ -16,11 +18,12 @@
+    * @return
     */
-   void flushGlobalACLCache();
+   CompletionStage<Void> flushGlobalACLCache();
```
This is public API, so we should deprecate rather than change it?

tristantarrant replied:
Not really: the only way to obtain a GlobalSecurityManager is through the component registry. I wouldn't worry about it

gustavonalle commented on server/rest/src/main/java/org/infinispan/rest/resources/SecurityResource.java:
```diff
@@ -80,9 +82,18 @@ public Invocations getInvocations() {
 
+   private CompletionStage<RestResponse> aclCacheFlush(RestRequest request) {
+      EmbeddedCacheManager cm = invocationHelper.getRestCacheManager().getInstance();
+      return SecurityActions.getGlobalComponentRegistry(cm).getComponent(GlobalSecurityManager.class).flushGlobalACLCache()
```
You can cache the ```GlobalSecurityManager``` instead of looking it up on every request I think

tristantarrant replied:
Will do

tristantarrant replied:
Rethinking about it: this is such a rare operation, that caching the GSM would be wasteful

gustavonalle commented on server/rest/src/main/java/org/infinispan/rest/resources/SecurityResource.java:
```diff
@@ -80,9 +82,18 @@ public Invocations getInvocations() {
             .handleWith(this::deny)
+            .invocation().methods(POST).path("/v2/security/cache").withAction("flush")
+            .permission(AuthorizationPermission.ADMIN).name("ACL CACHE FLUSH").auditContext(AuditContext.SERVER)
+            .handleWith(this::aclCacheFlush)
```
Not strictly related to this PR, but well, let's not waste the context :)
Could add an indent or a line break between each invocation declaration to make it easier to read?

gustavonalle commented on server/rest/src/main/java/org/infinispan/rest/resources/SecurityResource.java:
```diff
@@ -80,9 +82,18 @@ public Invocations getInvocations() {
+      return SecurityActions.getGlobalComponentRegistry(cm).getComponent(GlobalSecurityManager.class).flushGlobalACLCache()
+            .thenApply(v -> new NettyRestResponse.Builder().status(NO_CONTENT).build());
+   }
+
```
I couldn't find any documentation for the ```SecurityResource```, could you add it? It could be in another PR

oraNod commented:
@tristantarrant FYI I pushed some suggested doc for the ACL cache.

tristantarrant commented:
I've squashed the commits and added docs for the public security resource methods (I'd rather not document the methods which are of exclusive use to the console)

ryanemerson commented:
Thanks @tristantarrant
